app/components/TopCommunities/index.js

```diff
@@ -122,108 +125,206 @@ const TopCommunities = ({
-        {orderBy(communities, 'users_subscribed', 'desc')
-          .slice(0, 9)
-          .map(x => {
-            let Link = AStyled;
-            let route = routes.questions(x.id);
-            if (single && x.id !== single) {
-              Link = ADefaultStyled;
-              route = `${process.env.APP_LOCATION}${route}`;
-            } else if (single && x.id === single) {
-              route = routes.questions();
-            }
-
-            return (
-              <div key={x.id}>
-                <BaseRoundedNoPadding>
-                  <Link href={route} to={route}>
-                    <FrontSide>
-                      <div>
-                        <MediumImage src={x.avatar} alt="comm_img" />
-                        <P fontSize="16" bold>
-                          {x.name}
-                        </P>
-                      </div>
-
-                      <div>
-                        <div className="d-flex mb-3">
-                          <div className="d-flex flex-column flex-grow-1">
-                            <Span fontSize="16" bold>
-                              {getFormattedNum2(x.users_subscribed)}
-                            </Span>
-                            <Span
-                              className="mt-1"
-                              fontSize="14"
-                              color={TEXT_SECONDARY}
-                            >
-                              <FormattedMessage {...messages.users} />
-                            </Span>
-                          </div>
-                          <div className="d-flex flex-column flex-grow-1">
-                            <Span fontSize="16" bold>
-                              {getFormattedNum2(x.postCount)}
-                            </Span>
+  if (profile.ratings.length) {
+    return (
+      <div className="overlow-hidden" ref={ref}>
+        <H4 isHeader>
+          <FormattedMessage {...messages.myCommunities} />
+        </H4>
+
+        <Grid xl={5} lg={4} md={3} sm={2} xs={1}>
+          {orderBy(profile.ratings, 'rating', 'desc')
+            .slice(0, 9)
+            .map(x => {
+              let Link = AStyled;
+              let route = routes.questions(x.communityId);
+              if (single && x.communityId !== single) {
+                Link = ADefaultStyled;
+                route = `${process.env.APP_LOCATION}${route}`;
+              } else if (single && x.communityId === single) {
+                route = routes.questions();
+              }
+              const community = communities.find(
+                community => community.id == x.communityId,
+              );
+
+              return (
+                <div key={x.communityId}>
+                  <BaseRoundedNoPadding>
+                    <Link href={route} to={route}>
+                      <FrontSide>
+                        <div>
+                          <MediumImage src={community.avatar} alt="comm_img" />
```

Comment:
`community` can be undefined
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

please add a check if `community` exists, and then show the ui with this data